src/test/java/org/killbill/billing/plugin/adyen/api/TestAdyenPaymentPluginHttpErrors.java

```diff
@@ -159,6 +159,30 @@ public void testAuthorizeWithIncorrectValues() throws Exception {
         final PaymentTransactionInfoPlugin result = authorizeCall(account, payment, callContext, pluginApi, inValidPurchaseData);
 
         assertEquals(result.getStatus(), PaymentPluginStatus.ERROR);
+        assertEquals(result.getGatewayErrorCode(), "Expiry month should be between 1 and 12 inclusive");
+    }
+
+    /**
+     * Sanity check that a refused purchase (wrong cvv) still results in an error.
+     */
+    @Test(groups = "slow")
+    public void testAuthorizeWithIncorrectValuesCVV() throws Exception {
+
+        final Iterable<PluginProperty> inValidPurchaseData = invalidCreditCardDataCVV();
+
+        final Account account = defaultAccount();
+        final Payment payment = killBillPayment(account);
+        final AdyenCallContext callContext = newCallContext();
+
+        final AdyenPaymentPluginApi pluginApi = AdyenPluginMockBuilder.newPlugin()
+                                                                      .withAccount(account)
+                                                                      .withPayment(payment)
+                                                                      .build();
+
+        final PaymentTransactionInfoPlugin result = authorizeCall(account, payment, callContext, pluginApi, inValidPurchaseData);
+
+        assertEquals(result.getStatus(), PaymentPluginStatus.ERROR);
+        assertEquals(result.getGatewayErrorCode(), "CVC Declined");
```

Comment:
This doesn't really test the fix because the DAO is not involved (mock plugin). We should also verify `getPaymentInfo` is idempotent.